python/lsst/daf/butler/registry/_registry.py

```diff
@@ -657,84 +657,86 @@ def find(self, collection: str, datasetType: Union[DatasetType, str], dataId: Op
         return self._makeDatasetRefFromRow(result, datasetType=datasetType, dataId=dataId)
 
     @transactional
-    def addDataset(self, datasetType: Union[DatasetType, str],
-                   dataId: DataId, run: str, producer: Optional[Quantum] = None,
-                   recursive: bool = False, **kwds: Any) -> DatasetRef:
-        """Adds a Dataset entry to the `Registry`
+    def insertDatasets(self, datasetType: Union[DatasetType, str], dataIds: Iterable[DataId],
+                       run: str, *, producer: Optional[Quantum] = None, recursive: bool = False
+                       ) -> List[DatasetRef]:
+        """Insert one or more datasets into the `Registry`
```

Comment:
are dataIds themselves iterable? If not, despite the change in names, it might be nice to have this method take either a single dataId or Interable of DataIds, to prevent needing a bunch of trivial [dataId] all over the code. or maybe make a separate interface called InsertDataset that trivial defers to this method?

Reply:
Data IDs are indeed iterable (they're `collections.abc.Mapping`s).  Given that the actual high-level APIs are in `Butler`, and one of those (`put`) is already scalar rather than vector, let's see how many other places in the codebase would use `insertDataset` before adding it.